src/bottom-sheet.ts

```diff
@@ -28,8 +28,33 @@ type SheetUI = Readonly<{
     content: Array<SheetUIElement>;
 }>;
 
-export const bottomSheet = (payload: SheetUI): Promise<SheetResponse> =>
-    postMessageToNativeApp({type: 'SHEET', payload});
+let sheetLock = false;
+
+type SheetLockedResponse = {action: 'LOCKED'; result: Array<any>};
+
+export const bottomSheet = async (
+    payload: SheetUI,
+): Promise<SheetResponse | SheetLockedResponse> => {
+    if (sheetLock) {
+        return Promise.resolve({action: 'LOCKED', result: []});
+    }
```

Comment:
Wouldn't be better to reject this?

Reply:
CPS code is catching any error and showing the old picker in that case

Reply:
changed to an error